### PR TITLE
Minor fixes and adjustments

### PR DESCRIPTION
**Pull request by Jespertastesen**
file size now shows MB/GB instead of bytes
Added bootstrap theme
fixing allowed_ip implode/explode
fixing correct value of boolean
added allowed_ip as a "tag" in UI

V4B2cJfgeM

### CODE REVIEW

**Comment by Jespertastesen** [comment]
Annotation 2019-11-26 154440
Annotation 2019-11-26 154534

File: src/u/js/main.js [hunk 1]
```diff
@@ -30,17 +34,76 @@ function CopyKey() {
...
+    /* Get the text field */
+    var copyText = document.getElementById("secure_keyModal");
+
+    /* Select the text field */
+    copyText.select();
+    copyText.setSelectionRange(0, 99999); /*For mobile devices*/
+
+    /* Copy the text inside the text field */
+    document.execCommand("copy");
+}
```
Comment (Jespertastesen) [inline src/u/js/main.js]: Currently not working, need to fix.
Currently takes the first url always.

File: src/u/config.php [hunk 2]
```diff
@@ -3,5 +3,8 @@
 $settings = file_get_contents(__DIR__ . "/setup/settings.json");
 $settings = json_decode($settings, JSON_UNESCAPED_SLASHES);
 
+$version = ['version' => '1.0.0'];
```
Comment (JoeGandy) [inline src/u/config.php]: I'm not sure we should introduce version strings, I'd like to keep this project relatively simple :)

**Comment by Sossa07** [comment]
Is it possible to add to be able to have multiple accounts and all those accounts will have their own files. Like being able to have a personal account with your own files nobody can see unless they logged into your account.

**Comment by JoeGandy** [comment]
> Is it possible to add to be able to have multiple accounts and all those accounts will have their own files. Like being able to have a personal account with your own files nobody can see unless they logged into your account.

Possible yes, but I'd say we keep this out of this project, this is something that could be managed by the server, have a seperat config array for each person then handle that. Not something that would be part of the project. I have no plans to go down the rabbit hole of user auth

**Comment by Jespertastesen** [comment]
> > Is it possible to add to be able to have multiple accounts and all those accounts will have their own files. Like being able to have a personal account with your own files nobody can see unless they logged into your account.
> 
> Possible yes, but I'd say we keep this out of this project, this is something that could be managed by the server, have a seperat config array for each person then handle that. Not something that would be part of the project. I have no plans to go down the rabbit hole of user auth

Having a simple password protected site like this PR has, is easy enough, but making a multiuser auth, and possible different permissions would be imo too far for a simple file viewer like this.

There is some other out there that has that feature, but i wouldnt open my files up to some one else.
If i do, i would trust then enough to give them the master password anyway.

File: src/u/config.php (hunk 2 above)
Comment (Jespertastesen) [reply src/u/config.php]: removed in the latest PR

**Comment by Sossa07** [comment]
> > > Is it possible to add to be able to have multiple accounts and all those accounts will have their own files. Like being able to have a personal account with your own files nobody can see unless they logged into your account.
> > 
> > 
> > Possible yes, but I'd say we keep this out of this project, this is something that could be managed by the server, have a seperat config array for each person then handle that. Not something that would be part of the project. I have no plans to go down the rabbit hole of user auth
> 
> Having a simple password protected site like this PR has, is easy enough, but making a multiuser auth, and possible different permissions would be imo too far for a simple file viewer like this.
> 
> There is some other out there that has that feature, but i wouldnt open my files up to some one else.
> If i do, i would trust then enough to give them the master password anyway.

Link?

**Comment by Jespertastesen** [comment]
> > > > Is it possible to add to be able to have multiple accounts and all those accounts will have their own files. Like being able to have a personal account with your own files nobody can see unless they logged into your account.
> > > 
> > > 
> > > Possible yes, but I'd say we keep this out of this project, this is something that could be managed by the server, have a seperat config array for each person then handle that. Not something that would be part of the project. I have no plans to go down the rabbit hole of user auth
> > 
> > 
> > Having a simple password protected site like this PR has, is easy enough, but making a multiuser auth, and possible different permissions would be imo too far for a simple file viewer like this.
> > There is some other out there that has that feature, but i wouldnt open my files up to some one else.
> > If i do, i would trust then enough to give them the master password anyway.
> 
> Link?

@Sossa07 Check my latest 2 stars.

the 2. can be combined with this repo.
that one is databaseless.

**Comment by JoeGandy** [comment]
I think looking back at this PR, it does change everything too much, so I'm going to close it for now. I don't want to over complicate what should be a really simple project.

I would say if you're still keen to push this stuff, we can break it down and look at each feature, its just this PR has grown out of  the initial scope.

**Pull request closed without merging**